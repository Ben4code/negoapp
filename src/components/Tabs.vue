<template>
  <div class="tabs">
    <div class="tabs__header">
      <h4 class="tabs__control">Employer</h4>
      <h4 class="tabs__control">Employee</h4>
    </div>
    <div class="tabs__content">
      <Form/>
    </div>
  </div>
</template>

<script>
import Form from '@/components/Form'


export default {
  name: 'Tabs',
  components: {
    Form
  }
}
</script>

<style>
 .tabs{
  height: 70vh;
  margin: 5rem 15rem;
  background-color: #fff;
  box-shadow: 
    0 2px 2px 0 rgba(0,0,0,0.14), 
    0 3px 1px -2px rgba(0,0,0,0.12), 
    0 1px 5px 0 rgba(0,0,0,0.2);
 }
 .tabs__header{
    display: flex;
    justify-content: center;
    align-items: center;
 }
 .tabs__control{
   text-transform: uppercase;
    background: #00b8d4;
    padding: 1.5rem 1rem;
    width: 50%;
    text-align: center;
    cursor: pointer;
    color: #fff;
    box-shadow: 
      0 2px 2px 0 rgba(0,0,0,0.14), 
      0 3px 1px -2px rgba(0,0,0,0.12), 
      0 1px 5px 0 rgba(0,0,0,0.2);
 }
 .tabs__content{
   display: flex;
   justify-content: center;
   align-items: flex-start;
   padding-top: 5rem;
 }
</style>