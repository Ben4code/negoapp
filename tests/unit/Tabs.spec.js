import { shallowMount, mount } from '@vue/test-utils';
import Tabs from '@/components/Tabs'



let wrapper = null;

beforeEach(() => {
  wrapper = shallowMount(Tabs)
})

afterEach(() => {
  wrapper.destroy()
})

describe('Tabs Components Tests', () => {

  test('should mount component', () => {
    expect(wrapper.isVueInstance).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should display two tabs and correct text', () => {
    const tabsHandle = wrapper.findAll('.tabs__control');
    const EmployerTab = tabsHandle.at(0)
    const EmployeeTab = tabsHandle.at(1)
    expect(EmployerTab.text()).toBe('Employer');
    expect(EmployeeTab.text()).toBe('Employee');
    expect(tabsHandle.length).toBe(2);
  });

  test('should upadate active tab when toggleForm() is clicked', () => {
    const EmployerTab = wrapper.findAll('.tabs__control').at(0);
    EmployerTab.trigger('click');
    expect(wrapper.vm.state.formA).toBe(true);
    expect(wrapper.vm.state.formB).toBe(false);

    const EmployeeTab = wrapper.findAll('.tabs__control').at(1);
    EmployeeTab.trigger('click');
    expect(wrapper.vm.state.formA).toBe(false);
    expect(wrapper.vm.state.formB).toBe(true);
  });
});