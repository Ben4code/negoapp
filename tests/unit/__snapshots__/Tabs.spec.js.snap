// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Components Tests should mount component 1`] = `
<div class="tabs">
  <div class="tabs__header">
    <h4 class="active tabs__control">Employer</h4>
    <h4 class="tabs__control">Employee</h4>
  </div>
  <div class="tabs__content">
    <form-stub formdata="[object Object]"></form-stub>
  </div>
</div>
`;
